tests/daemon/test-machine-id-provider.c

```diff
@@ -125,73 +146,52 @@ test_machine_id_provider_new_succeeds (MachineIdTestFixture *fixture,
-                                     gconstpointer         dontuseme)
-{
-  g_autoptr(EmerMachineIdProvider) id_provider =
-    emer_machine_id_provider_new_full (fixture->machine_id_file_path,
-                                       fixture->override_machine_id_file_path);
-  uuid_t id;
-  g_assert (emer_machine_id_provider_get_id (id_provider, id));
-  gchar unparsed_id[HYPHENS_IN_ID + FILE_LENGTH];
-  uuid_unparse_lower (id, unparsed_id);
-  g_autofree gchar* unhypenated_id = unhyphenate_uuid (unparsed_id);
-  g_assert_cmpstr (TESTING_ID, ==, unhypenated_id);
-}
-
-static void
-test_machine_id_provider_can_get_id_override (MachineIdTestFixture *fixture,
+test_machine_id_provider_can_get_tracking_id (MachineIdTestFixture *fixture,
                                               gconstpointer         dontuseme)
 {
   g_autoptr(EmerMachineIdProvider) id_provider =
-    emer_machine_id_provider_new_full (fixture->machine_id_file_path,
-                                       fixture->override_machine_id_file_path);
+    emer_machine_id_provider_new_full (fixture->tracking_id_file_path);
   uuid_t id;
-  write_testing_machine_id (fixture->override_machine_id_file_path, TESTING_OVERRIDE_ID);
+  write_testing_machine_id (fixture->tracking_id_file_path, TESTING_TRACKING_ID);
   g_assert (emer_machine_id_provider_get_id (id_provider, id));
   gchar unparsed_id[HYPHENS_IN_ID + FILE_LENGTH];
   uuid_unparse_lower (id, unparsed_id);
   g_autofree gchar* unhypenated_id = unhyphenate_uuid (unparsed_id);
-  g_assert_cmpstr (TESTING_OVERRIDE_ID, ==, unhypenated_id);
+  g_assert_cmpstr (TESTING_TRACKING_ID, ==, unhypenated_id);
 }
 
 static void
-test_machine_id_provider_writes_correctly_formed_override_id (MachineIdTestFixture *fixture,
+test_machine_id_provider_writes_correctly_formed_tracking_id (MachineIdTestFixture *fixture,
                                                               gconstpointer         dontuseme)
 {
   g_autoptr(EmerMachineIdProvider) id_provider =
-    emer_machine_id_provider_new_full (fixture->machine_id_file_path,
-                                       fixture->override_machine_id_file_path);
+    emer_machine_id_provider_new_full (fixture->tracking_id_file_path);
   g_autoptr(GError) error = NULL;
   g_autofree gchar *contents = NULL;
 
   emer_machine_id_provider_reset_tracking_id (id_provider, &error);
   g_assert_no_error (error);
 
-  /* Read the override_machine_id_file_path using g_file_get_contents
+  /* Read the tracking_id_file_path using g_file_get_contents
    * and check that its size matches what we would normally write to the
    * file */
-  g_file_get_contents (fixture->override_machine_id_file_path, &contents, NULL, &error);
+  g_file_get_contents (fixture->tracking_id_file_path, &contents, NULL, &error);
   g_assert_no_error (error);
 
-  g_assert_cmpint (strlen (contents), ==, strlen (TESTING_OVERRIDE_ID));
+  g_assert_cmpint (strlen (contents), ==, strlen (TESTING_TRACKING_ID));
```

Comment:
I know this is pre-existing code, but we should assert that the new ID is not the same as the old ID. (There is a 2 ** 128 chance of this failing but that's OK.)

We should also call `emer_machine_id_provider_get_id` and check that it returns a different ID.

Reply:
addressed